### PR TITLE
Pre-commit integration

### LINKED ISSUE #14: Configure pre-commit tool for this repo
> # Rules:
> 1. **DO NOT** commit to master. ever.
> 2. **ABSOLUTELY NO** force pushes allowed for public non-feature branches.
> 3. For every feature, create a new branch matching name `feature/<issue-number>-<feature-name>`.
> 4. Submit every feature by creating Pull Request @ github, after that wait for review, fix issues, repeat until approved. If needed rebase the branch on top of master and re-push force it.

  * [x] **I HAVE READ AND UNDERSTOOD THE RULES ABOVE**
------------

* [x] Install pre-commit locally as follows:
```
python3.7 -m pip install pre-commit --user
python3.7 -m pre_commit install
```
* [x] Create branch called `feature/pre-commit-integration`
* [x] Add [`.pre-commit-config.yaml`](http://pre-commit.com/) config, containing the following [hooks](http://pre-commit.com/hooks.html):
  * [x] no-commit-to-branch
  * [x] [add-trailing-comma](https://github.com/asottile/add-trailing-comma)
  * [x] [remove-tabs](https://github.com/Lucas-C/pre-commit-hooks)
  * [x] trailing-whitespace
  * [x] check-merge-conflict
  * [x] double-quote-string-fixer
  * [x] requirements-txt-fixer
  * [x] check-yaml
  * [x] check-docstring-first
  * [x] end-of-file-fixer
  * [x] name-tests-test
  * [x] check-ast
  * [x] debug-statements
  * [x] flake8
  * [x] [pydocstyle](https://github.com/chewse/pre-commit-mirrors-pydocstyle)
  * [x] `pylint` using "local" mode
  * [x] [mypy](https://github.com/pre-commit/mirrors-mypy.git)
* [x] Check repo using command `pre-commit run --all-files`, fix all issues and commit fixes as well
* [x] Create PR with this branch

### PR DESCRIPTION
**Pull request by RootenberG**
Closes #14

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  - id: pydocstyle
+    name: Verifying PEP257 Compliance
+    entry: pydocstyle
+    language: python
+    files: \.py$
+- repo: local
+  hooks:
+  - id: pylint
+    name: pylint
+    entry: python -m pylint.__main__
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: This should be enough. Read more about `__main__` package entrypoints.
```suggestion
    entry: python3 -m pylint
```

**Comment by webknjaz** [comment]
Squash commits, because the message on the second one is useless

**Comment by webknjaz** [comment]
Did you close it again?

**Comment by RootenberG** [comment]
My bad, I'll fix it.

**Comment by webknjaz** [comment]
I see you like practicing your Git-fu

**Comment by RootenberG** [comment]
I can't reopen this PR, even when I return to the commit that was before closing the PR.

**Comment by webknjaz** [comment]
Yes, because you merged it.

**Comment by RootenberG** [comment]
So, is it impossible now?

**Comment by webknjaz** [comment]
Exactly

**Comment by RootenberG** [comment]
Can I open a new PR? The last.

**Comment by webknjaz** [comment]
Sure, no prob. Open as many PRs as you need. It's your playground and it's okay to try out different things :)

**Pull request closed without merging**